src/api/app/services/diff_parser.rb

```diff
@@ -49,8 +49,10 @@ def generate_inline_diffs
         sign = line.state == 'added' ? '+' : '-'
 
         # Known scenario https://github.com/openSUSE/open-build-service/pull/17006
-        # Log if a different unexpected scenario happen instead
-        Rails.logger.warn "generate_inline_diffs: content is nil - line='#{line.to_json}'" if content.nil? && line.content != '+'
+        # Raise exception if a different unexpected scenario happen instead
+        if content.nil? && line.content != '+'
+          raise "[Unknown scenario in generate_inline_diffs: content is nil but line is not '+': line='#{line.to_json}'"
```

Comment:
Raising something means execution stops. That's not something you want to do right? Are you looking for `Airbrake.notify`?